include/eixx/util/atom_table.hpp

```diff
@@ -103,58 +110,56 @@ namespace util {
         }
 
         /// Lookup an atom in the atom table by index.
-        const String& get(int n) const { return (*this)[n]; }
+        const String& get(size_t n) const { return (*this)[n]; }
 
         /// Lookup an atom in the atom table by index.
-        const String& operator[] (int n) const {
+        const String& operator[] (size_t n) const {
             BOOST_ASSERT((size_t)n < m_atoms.size());
             return m_atoms[n];
         }
 
         /// Try to lookup an atom in the atom table
         /// @return -1 if the atom by name is not found, -2 if the atom is invalid,
         ///         or a value >= 0, if an existing atom is found.
-        int try_lookup(const char* a_name, size_t n) { return try_lookup(String(a_name, n)); }
-        int try_lookup(const char* a_name)           { return try_lookup(String(a_name)); }
-        int try_lookup(const String& a_name)
+        size_t try_lookup(const char* a_name, size_t n) { return try_lookup(String(a_name, n)); }
+        size_t try_lookup(const char* a_name)           { return try_lookup(String(a_name)); }
+        size_t try_lookup(const String& a_name)
         {
```

Comment:
Values that are not found works as expected due to (size_t)-1 casting.

Reply:
With this change, the user of this function would need to write something like:
```
auto res = some_atom_table().try_lookup("abc");
if (ssize_t(res) < 0)
  ...
```
Which is a rather old API that they need to typecast the return.  Alternatively, the return type can be changed to:
```
std::pair<bool, size_t> try_lookup(...)
```
where the first item in the pair would be the success status of the lookup, and the second would be what the function currently returns.  This would be a cleaner API than requiring the user to do a typecasted check.

Reply:
The other alternative would be to define some static constants in the table:
```
static const size_t npos = size_t(-1);
static const size_t name_too_long = size_t(-2);
```
And then the API user would do:
auto res = some_atom_table().try_lookup("abc");
if (res != atom_table::npos && res != atom_table::name_too_long)
  // success
```
But that is also too verbose.  So, probably the `std::pair` approach would be the better API choice.

Reply:
Agreed. `std::pair` approach would be the better API choice.

Reply:
Fixed in latest commit https://github.com/saleyn/eixx/pull/50/commits/506b26d4427be7a0c1feb2f029f601650954cdc2